### PR TITLE
usafe component will receive props

### PR DESCRIPTION
**Pull request by benitogf**
#534

### CODE REVIEW

**Comment by MaxSvargal** [comment]
Let's refactor it to `static getDerivedStateFromProps`

**Comment by benitogf** [comment]
Hello @MaxSvargal made some progress on using `static getDerivedStateFromProps` however got blocked on https://github.com/oliviertassinari/react-swipeable-views/pull/538/files#diff-763c63c1eb389e9b13f4cc3e2eff08abR354 as I mentioned in the comment this method relies on having access to the class `this` which is not available from `getDerivedStateFromProps` any suggestions?

File: packages/react-swipeable-views/src/SwipeableViews.js [hunk 1]
```diff
@@ -359,6 +343,32 @@ class SwipeableViews extends React.Component {
...
+  static getDerivedStateFromProps(nextProps, prevState) {
+    const { index, children } = nextProps;
+
+    if (typeof index === 'number' && index !== prevState.indexLatest) {
+      if (process.env.NODE_ENV !== 'production') {
+        checkIndexBounds(nextProps);
+      }
+
+      // this.setIndexCurrent(index);
+      // this method heavily realies on having access to the class :/
```
Comment (piotr-cz) [inline packages/react-swipeable-views/src/SwipeableViews.js]: realies -> relies

**Comment by jeanpoelie** [comment]
@benitogf Can you checkout the unit test that failed, it returned the wrong transform. could you also fix the typo in the comments?

**Comment by benitogf** [comment]
@jeanpoelie will take a look to this one on the weekend 👍 could we merge the UNSAFE_componentWillReceiveProps while we switch to getDerivedStateFromProps though?

**Comment by jeanpoelie** [comment]
I went ahead and merged #551 after resolving the conflicts.

**Comment by jeanpoelie** [comment]
@benitogf did you find the time to take a look at this?

**Comment by benitogf** [comment]
Hello, I was trying to take a look at this one but found that currently the class has both [unsafe labeled](https://github.com/oliviertassinari/react-swipeable-views/blob/master/packages/react-swipeable-views/src/SwipeableViews.js#L705) and [regular](https://github.com/oliviertassinari/react-swipeable-views/blob/master/packages/react-swipeable-views/src/SwipeableViews.js#L322) methods is there a reason for that? shouldn't just the unsafe call be ok?

about the issue itself, found that [this.indexCurrent = indexCurrent](https://github.com/oliviertassinari/react-swipeable-views/blob/master/packages/react-swipeable-views/src/SwipeableViews.js#L360) allows switching between a "controled" and "uncontrolled" state, moving forward I think that the component should be fully controled after the initial render, changing the state acording to prop changes is an [antipattern](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-controlled-component)?  Just sharing what I have found so far, glad to get other ideas about it

**Pull request merged**